cmd/frontend/graphqlbackend/search.go

```
@@ -342,24 +342,36 @@ func decodedViewerFinalSettings(ctx context.Context) (*schema.Settings, error) {
 	return &settings, nil
 }
 
-var mockResolveRepoGroups func() (map[string][]*types.Repo, error)
+var mockResolveRepoGroups func() (map[string][]*types.Repo, error, []string)
 
-func resolveRepoGroups(settings *schema.Settings) (map[string][]*types.Repo, error) {
+type regexPath struct {
+	name  string
+	regex string
+}
+
+func resolveRepoGroups(settings *schema.Settings) (map[string][]*types.Repo, error, []string) {
 	if mockResolveRepoGroups != nil {
 		return mockResolveRepoGroups()
 	}
-
 	groups := map[string][]*types.Repo{}
+	var patterns []string
 
 	for name, repoPaths := range settings.SearchRepositoryGroups {
-		repos := make([]*types.Repo, len(repoPaths))
-		for i, repoPath := range repoPaths {
-			repos[i] = &types.Repo{Name: api.RepoName(repoPath)}
+		repos := make([]*types.Repo, 0, len(repoPaths))
+
+		for _, repoPath := range repoPaths {
+			if repoPathString, ok := repoPath.(string); ok {
```

I'd recommend using a type switch here. You should also return an error if you get an unexpected type here.
```
switch v := repoPath.(type) {
  case string:
    ...
  case map[string]interface{}:
    ...
  default:
    return error
}
```